### PR TITLE
Weights & Biases Hyperparameter Optimization

### PR DESCRIPTION
**Pull request by nrafaili**
This pull request integrates W&B hyperparameter optimization to Protify.

The way it works is:

If you want to run hyperparameter optimization after passing a set of `--model_names` and ``--data_names`` (or ``--data_dirs``) you can add to your CLI ``--use_wandb_hyperopt``. It works with all transfer learning options.

There are a number of arguments you can choose to specify, such as what metric to sweep over, what method (bayes, random, grid), and how many trials to conduct per sweep, among others.

When Random or Random-Transformer has been added to `--model_names``, hyperparameter optimization will not occur for either of those and a regular training run will be conducted with those embeddings.

### CODE REVIEW

**Comment by lhallee** [comment]
A couple things here @nrafaili 

Make sure any of the random models and one hot models are not subject to the tuning.

It would be great if we could add any of the probe choices, linear, transformer, lyra, full finetuning, linear then hyprid + lora, etc. There are a bunch of options. It would be cool if the optimization would search over THOSE and their internal hypers, most importantly hidden size and number of layers. We should automatically scale things like the attention heads so that we don't have to ever search over those.

Is that currently possible with this schema? There are difficulties with this because like, it is reasonable to have a hidden size of 16000+ for a linear model but absolutely not for a transformer model, lol. Give this some thought.

But, I think the dream here would be to have a new dataset, send in a query for random, random transformer, one hot, and ESMC-300 (or our newer CAMP models, whatever), and then the hyper parameter optimization finds a near optimal solution of probe, size, and training scheme to get the best test score.
